src/DataLayouts/cuda.jl

```diff
@@ -19,9 +19,13 @@ Adapt.adapt_structure(to, data::IF{S, Ni}) where {S, Ni} =
 Adapt.adapt_structure(to, data::VF{S}) where {S} =
     VF{S}(Adapt.adapt(to, parent(data)))
 
-parent_array_type(
-    ::Type{A},
-) where {A <: CUDA.GPUArrays.AbstractGPUArray{FT}} where {FT} = CUDA.CuArray{FT}
+parent_array_type(::Type{<:CUDA.GPUArrays.AbstractGPUArray{T}}) where {T} =
+    CUDA.CuArray{T}
+
+promote_container_type(
+    ::Type{<:CUDA.GPUArrays.AbstractGPUArray{T1}},
```

Comment:
this needs to have a condition that the dimensions and buffers are the same type:
```
::GPUArray{T1, Dim, Buffer}
::GPUArray{T2, Dim, Buffer}
where {T1, T2, Dim, Buffer}
```